terraform/staging/main.tf

```
@@ -93,6 +97,10 @@ provider "github" {
   token = jsondecode(data.aws_secretsmanager_secret_version.flux_github_token.secret_string).flux_github_token
 }
 
+provider "gitlab" {
+  base_url = "https://gitlab.staging.spack.io"
+  token = jsondecode(data.aws_secretsmanager_secret_version.gitlab_token.secret_string).gitlab_terraform_provider_access_token
+}
```

I'm not suggesting that we need to do this since it might be an overcomplication, but I was playing around with this and found out you can actually extract the gitlab url from the cluster.
```suggestion
data "kubernetes_ingress_v1" "gitlab_webservice" {
  metadata {
    name      = "gitlab-webservice-default"
    namespace = "gitlab"
  }
}

locals {
  gitlab_url = "https://${data.kubernetes_ingress_v1.gitlab_webservice.spec[0].rule[0].host}"
}

provider "gitlab" {
  base_url = local.gitlab_url
  token = jsondecode(data.aws_secretsmanager_secret_version.gitlab_token.secret_string).gitlab_terraform_provider_access_token
}
```